map/framework.cpp

```diff
@@ -2448,26 +2448,39 @@ bool Framework::ParseEditorDebugCommand(search::SearchParams const & params)
 
 namespace
 {
-vector<osm::LocalizedStreet> FilterNearbyStreets(vector<search::ReverseGeocoder::Street> const & streets,
-                                                 string const & exactFeatureStreet = "")
+osm::LocalizedStreet LocalizeStreet(model::FeaturesFetcher const & model, FeatureID const fid)
+{
+  osm::LocalizedStreet result;
+  Index::FeaturesLoaderGuard g(model.GetIndex(), fid.m_mwmId);
+  FeatureType ft;
+  g.GetOriginalFeatureByIndex(fid.m_index, ft);
+  ft.GetPreferredNames(result.m_defaultName, result.m_localizedName);
+  return result;
+}
+
+vector<osm::LocalizedStreet> TakeSomeStreetsAndLocalize(
+    vector<search::ReverseGeocoder::Street> const & streets,
+    model::FeaturesFetcher const & model)
+
 {
   vector<osm::LocalizedStreet> results;
   // Exact feature street always goes first in Editor UI street list.
 
-  // TODO: Push into result LocalizedStreet object with default and localized street name.
-  if (!exactFeatureStreet.empty())
-    results.push_back({exactFeatureStreet, ""});
   // Reasonable number of different nearby street names to display in UI.
   constexpr size_t kMaxNumberOfNearbyStreetsToDisplay = 8;
   for (auto const & street : streets)
   {
-    osm::LocalizedStreet const st{street.m_name, ""};
-    if (find(results.begin(), results.end(), st) == results.end())
-    {
-      results.push_back(st);
-      if (results.size() >= kMaxNumberOfNearbyStreetsToDisplay)
-        break;
-    }
+    auto const isDuplicate = find_if(begin(results), end(results),
+                                     [&street](osm::LocalizedStreet const & s)
+                                     {
+                                       return s.m_defaultName == street.m_name;
```

Comment:
Хм, для чего?

> On 14.4.2016, at 10.53, mgsergio <notifications@github.com> wrote:
> 
> In map/framework.cpp:
> 
> >    // Reasonable number of different nearby street names to display in UI.
> >    constexpr size_t kMaxNumberOfNearbyStreetsToDisplay = 8;
> >    for (auto const & street : streets)
> >    {
> > -    osm::LocalizedStreet const st{street.m_name, ""};
> > -    if (find(results.begin(), results.end(), st) == results.end())
> > -    {
> > -      results.push_back(st);
> > -      if (results.size() >= kMaxNumberOfNearbyStreetsToDisplay)
> > -        break;
> > -    }
> > +    auto const isDuplicate = find_if(begin(results), end(results),
> > +                                     [&street](osm::LocalizedStreet const & s)
> > +                                     {
> > +                                       return s.m_defaultName == street.m_name;
> Используется в editor_dialog
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>